src/boot/util.h

```diff
@@ -100,6 +100,13 @@ static inline Pages xmalloc_pages(
 }
 
 static inline Pages xmalloc_initrd_pages(size_t n_pages) {
+        /* The original native x86 boot protocol of the Linux kernel was not 64bit safe, hence we allocate
+         * memory for the initrds below the 4G boundary on x86, since we don't know early enough which
+         * protocol we'll use to ultimately boot the kernel. This restriction is somewhat obsolete, since
+         * these days we generally prefer the kernel's newer EFI entrypoint instead, which has no such
```

Comment:
> the kernel's newer EFI entrypoint instead, which has no such limitations

Incidentally, the kernel's EFI entrypoint [had once been patched](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=47226ad4f4cfd1e91ded7f2ec42f83ff1c624663) to deliberately limit initrd loading to below 4G because my laptop's UEFI firmware would crash otherwise.

(I no longer use that laptop, and AFAIK all of that code got removed during a ~2020 refactor of the stub code.)